alacritty/src/display/mod.rs

```diff
@@ -365,23 +364,26 @@ pub struct Display {
     /// The ime on the given display.
     pub ime: Ime,
 
-    // OpenGL context.
-    context: Replaceable<PossiblyCurrentContext>,
-
-    // OpenGL surface.
-    surface: Surface<WindowSurface>,
-
     // Renderer.
     renderer: Renderer,
 
     // Mouse point position when highlighting hints.
     hint_mouse_point: Option<Point>,
 
+    surface: Surface<WindowSurface>,
+
+    // XXX OpenGL context must be dropped after the renderer.
+    context: Replaceable<PossiblyCurrentContext>,
+
     debug_damage: bool,
     damage_rects: Vec<DamageRect>,
     next_frame_damage_rects: Vec<DamageRect>,
     glyph_cache: GlyphCache,
     meter: Meter,
+
+    // XXX window should be dropped in the end, otherwise `surface` and other GL operations could
+    // have issues during freeing resources.
+    pub window: Window,
```

Comment:
I'd definitely place all the things that require manual dropping at the end together. Splitting it up like this seems like a recipe for disaster.

This seems fairly complicated so I'd be fine with a manual `Drop` impl too.